src/cunumeric/scan/scan_local_template.inl

```diff
@@ -0,0 +1,107 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "cunumeric/scan/scan_local_util.h"
+#include "cunumeric/pitches.h"
+
+namespace cunumeric {
+
+using namespace Legion;
+using namespace legate;
+
+template <VariantKind KIND, ScanCode OP_CODE, LegateTypeCode CODE, int DIM>
+struct ScanLocalImplBody;
+
+template <VariantKind KIND, ScanCode OP_CODE, LegateTypeCode CODE, int DIM>
+struct ScanLocalNanImplBody;
+
+template <VariantKind KIND, ScanCode OP_CODE, bool NAN_TO_IDENTITY>
+struct ScanLocalImpl {
+  // NOTE: currently NAN conversion is not supported for complex types, once implemented the
```

Comment:
this line doesn't seem consistent with what the code is doing; isn't this template instantiated when `NAN_TO_IDENTITY` is `true` and the type is a complex type?

Reply:
Yea, forgot to remove it after fixing isnan.
9d3e545ec7d6f137e937c0b48b1c256b8721a4e0